src/fps.rs

```diff
@@ -0,0 +1,59 @@
+use std::future::Future;
+use std::sync::atomic::Ordering;
+use std::time::Duration;
+
+use dcs_module_ipc::IPC;
+use stubs::mission::v0::stream_events_response::{Event, FpsEvent};
+use stubs::mission::v0::StreamEventsResponse;
+use tokio::time::{interval, MissedTickBehavior};
+
+use crate::{FPS, TIME};
+
+pub async fn run_in_background(
+    ipc: IPC<StreamEventsResponse>,
+    mut shutdown_signal: impl Future<Output = ()> + Unpin,
+) {
+    let mut interval = interval(Duration::from_secs(1));
+    interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+
+    // clear FPS counter when first being started
+    let mut previous = interval.tick().await;
+    FPS.store(0, Ordering::Relaxed);
+
+    loop {
+        // wait for either the shutdown signal or the next interval tick, whatever happens first
+        let instant = tokio::select! {
+            _ = &mut shutdown_signal => {
+                break
+            }
+            instant = interval.tick() => instant
+        };
+
+        // Technically, there could be a simulation frame between the read of `frame_count` and
+        // `time` so that `time` already receives a newer value. However, this should be rare and
+        // shouldn't really matter for the resolution measured here.
+        let frame_count = FPS.swap(0, Ordering::Relaxed);
+        let time = TIME.load(Ordering::Relaxed);
+
+        let elapsed = instant - previous;
+        previous = instant;
+        let fps = (frame_count as f64) / elapsed.as_secs_f64();
+
+        ipc.event(StreamEventsResponse {
+            time: f64::from(time) / 1000.0,
+            event: Some(Event::Fps(FpsEvent { fps })),
+        })
+        .await;
+
+        ipc.request::<_, stubs::trigger::v0::OutTextResponse>(
+            "outText",
+            Some(stubs::trigger::v0::OutTextRequest {
+                text: format!("Server FPS: {:.2}", fps),
+                display_time: 2,
+                clear_view: true,
+            }),
+        )
+        .await
+        .ok();
```

Comment:
```suggestion
```

Needs to be removed before merge. Just in here to ease testing. I've compared its output with DCS' FPS counter and Nividia's performance overlay and seem to match up close enough.